### PR TITLE
feat(ci): cancel in progress CI if there is new commit in a same PR

### PR DESCRIPTION
**Pull request by FrankYang0529**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

**Solution:**

**Related Issue:**
https://github.com/harvester/harvester/issues/6711

**Test plan:**

### CODE REVIEW

**Comment by FrankYang0529** [comment]
Cancel in progress CI result: https://github.com/harvester/harvester/actions/runs/11322270751

**Comment by bk201** [comment]
Will the change cancel the running job in another PR?

**Comment by FrankYang0529** [comment]
> Will the change cancel the running job in another PR?

No, it uses PR number as name or `girhub.ref`. For branch CI, we skip canceling in progress CI.

**Review by bk201 — APPROVED** [review]

**Review by tserong — APPROVED** [review]
Nice!

**Comment by tserong** [comment]
I know I just approved this, but I subsequently realised that we'll need to eventually update `cancel-in-progress: ${{ !contains(fromJson('["master", "v1.2", "v1.3", "v1.4", "v1.5"]'), github.ref) }}` once we get to harvester v1.6 (and v1.7, and ...).

It might be interesting to experiment with something like `github.ref != "master" && !startsWith(gihub.ref, "v")` to exclude master and all v-named branches instead. 

...and it might need to be `github.ref_name` which is the short ref name, vs `github.ref` which will be something like `refs/heads/...` according to https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables

**Comment by FrankYang0529** [comment]
Hi @tserong, updated it. Thanks for the suggestion.

**Review by tserong — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -10,6 +10,9 @@ on:
   pull_request:
   schedule:
     - cron: "10 0,12 * * *"
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' && !startsWith(github.ref, 'refs/tags/v') }}
```
Comment (bk201) [inline .github/workflows/build.yml]: Does this work for v1.x branches?

Comment (tserong) [reply .github/workflows/build.yml]: Oooh, good point, that means I think it'll need to be something like:

```
${{ github.ref != 'refs/heads/master' && !startsWith(github.ref, 'refs/heads/v') && !startsWith(github.ref, 'refs/tags/v') }}
```

That should cover both branches and tags.

Comment (FrankYang0529) [reply .github/workflows/build.yml]: Updated it. Thanks.

**Review by tserong — APPROVED** [review]

**Review by bk201 — APPROVED** [review]

**Pull request merged**